core/src/main/scala/chisel3/Bits.scala

```diff
@@ -304,10 +310,14 @@ sealed abstract class Bits(private[chisel3] val width: Width) extends Element wi
   def do_>> (that: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bits
 
   /** Returns the contents of this wire as a [[scala.collection.Seq]] of [[Bool]]. */
-  final def toBools(): Seq[Bool] = macro SourceInfoTransform.noArg
+  final def toBools: Seq[Bool] = macro SourceInfoTransform.noArg
+  @deprecated("Calling this function with an empty argument list is invalid in Scala 3. Use the form without parentheses instead", "3.5")
+  final def toBools(dummy: Int*): Seq[Bool] = macro SourceInfoWhiteboxTransform.noArgDummy
```

Comment:
Interestingly, `.toBools` is supposed to be gone and I'm a bit confused why it's here... Definitely not the fault of this PR but weird.

Reply:
I think its still only a deprecation warning?

Reply:
EDIT: seems like it still exists for `Bits`: https://github.com/chipsalliance/chisel3/blob/4b7b771eeced366345779a75987ce552558a1c7e/core/src/main/scala/chisel3/Bits.scala#L307
Should we get rid of that?

Reply:
oh. This is the line you are talking about 😂 . Yeah lets get rid of this one in another PR

Reply:
> Interestingly, .toBools is supposed to be gone and I'm a bit confused why it's here... Definitely not the fault of this PR but weird.

My bad, I think did didn't full remove it in https://github.com/chipsalliance/chisel3/pull/1730

I think these should be removed, I can create another PR, or @jared-barocsi can do it in this PR.

Reply:
@sequencer please create another PR (I opened #2143 to track), as it's not really related to this PR